### PR TITLE
Notebooks and consoles should swallow all Escape keydown events.

### LINKED ISSUE #282: `esc` shouldn't be caught in the terminal to close the file manager
Right now, `esc` is caught when the terminal is open and always closes the file manager. The problem is that `esc` is often a necessary sequence in the terminal (vi, emacs, etc.), so we need at least the terminal to get all `esc` events unmodified.

### PR DESCRIPTION
**Pull request by afshin**
Fixes #282.

### CODE REVIEW

File: src/console/tooltip.ts [hunk 1]
```diff
@@ -99,25 +99,21 @@ class ConsoleTooltip extends Widget {
    * Handle `after_attach` messages for the widget.
    *
    * #### Notes
-   * Captures window events in capture phase to dismiss the tooltip widget.
-   *
-   * Because its parent (reference) widgets use window listeners instead of
-   * document listeners, the tooltip widget must also use window listeners
-   * in the capture phase.
+   * Captures window events in to dismiss the tooltip widget.
```
Comment (blink1073) [inline src/console/tooltip.ts]: Needs reword.

**Comment by blink1073** [comment]
Also need to upgrade phosphide in the `lab` example.

**Comment by blink1073** [comment]
Escape is no longer converting to "command" mode in the notebook.

**Comment by ellisonbg** [comment]
I think it is too aggressive for these plugins to shallow all esc events.
There are lots of different plugins that might use esc.

On Mon, Jul 11, 2016 at 1:53 PM, Steven Silvester <notifications@github.com>
wrote:

> Escape is no longer converting to "command" mode in the notebook.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jupyter/jupyterlab/pull/290#issuecomment-231829430>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABr0EDEhoYowimOeYa2gXmv8d4kSzE0ks5qUpERgaJpZM4JJqWO>
> .
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by afshin** [comment]
@ellisonbg They only swallow escape keydown events if they have the focus, not in all situations. The idea is that escape is a special key that means something to editors/notebooks, so it shouldn't bubble up. If you hit escape once to go into notebook command mode, the next time should not accidentally dismiss the command palette.

**Comment by blink1073** [comment]
Nice!

**Pull request merged**